src/license/license.proto

```diff
@@ -0,0 +1,109 @@
+syntax = "proto3";
+
+package license;
+option go_package = "github.com/pachyderm/pachyderm/v2/src/license";
+
+import "google/protobuf/timestamp.proto";
+import "gogoproto/gogo.proto";
+import "enterprise/enterprise.proto";
+
+message ActivateRequest {
+  // activation_code is a Pachyderm enterprise activation code. New users can
+  // obtain trial activation codes
+  string activation_code = 1;
+
+  // expires is a timestamp indicating when this activation code will expire.
+  // This should not generally be set (it's primarily used for testing), and is
+  // only applied if it's earlier than the signed expiration time in
+  // 'activation_code'.
+  google.protobuf.Timestamp expires = 2;
+}
+
+message ActivateResponse {
+  enterprise.TokenInfo info = 1;
+}
+
+message GetActivationCodeRequest {}
+
+message GetActivationCodeResponse {
+  enterprise.State state = 1;
+  enterprise.TokenInfo info = 2;
+  string activation_code = 3;
+}
+
+message DeactivateRequest{}
+message DeactivateResponse{}
+
+message AddClusterRequest {
+  // id is the unique, immutable identifier for this cluster
+  string id = 1;
+  // address is the public GPRC address where the cluster can be reached
```

Comment:
> Just checking my understanding of the design here: I know we use the address when enrolling a new pach cluster, to check that its hostname matches the hostname in the license. Is that the only use or do we use it for other stuff too?

This is the hostname of the pachd, not the hostname of the license server. The pachd hostnames aren't in the license at all. This is mostly for administrative purposes (you get the hostname back when you list clusters), and so that we can run commands on "managed" pachds from the enterprise server.

> Also, this might just by ignorance of common security practices on my part, but why would one want to impose a constraint on hostname in the license? (vs. e.g. limiting the total number of clusters). Is it to prevent like an unauthorized cluster from sneaking onto someone's license server?

We don't impose a constraint on the pachd hostname in the license. We may impose a constraint on the license server hostname, which is just a measure to make it harder to reuse the same license on many servers.